docs/client-libraries-schema.md

```diff
@@ -142,4 +142,96 @@ while True:
     except Exception:
         # Message failed to be processed
         consumer.negative_acknowledge(msg)
-```
\ No newline at end of file
+```
+
+## Work with the Go schema
+Working with Go schema has slight differences from Java schema.
+This part will introduce the schema compatibility between Go client and Java client.
+
+#### Avro Schema
+Avro schema in Go and Java are compatible, but there are some differences in how the schemas are defined. Go typically uses schema definitions, a string JSON, to create schemas. However, Java often uses class types for schema creation. As a result, Java allows non-primitive fields to be nullable by default, while in Go's schema definition, the nullability of fields needs to be explicitly stated.
+GO:
+```go
+// Compatible with defining a schema in Java
+exampleSchemaDefCompatible := NewAvroSchema(`{"fields":
+    [
+        {"name":"id","type":"int"},{"default":null,"name":"name","type":["null","string"]}
+    ],
+    "name":"MyAvro","namespace":"schemaNotFoundTestCase","type":"record"}`, nil)
+// Not compatible with defining a schema in Java
+exampleSchemaDefIncompatible := NewAvroSchema(`{"fields":
+    [
+        {"name":"id","type":"int"},{"default":null,"name":"name","type":["string"]}
+    ],
+    "name":"MyAvro","namespace":"schemaNotFoundTestCase","type":"record"}`, nil)
+Producer := NewAvroSchema(exampleSchemaDef, nil)
+
+```
+JAVA:
+```java
+@AllArgsConstructor
+@NoArgsConstructor
+public static class Example {
+    public String name;
+    public int id;
+}
+
+Producer<Example> producer = pulsarClient.newProducer(Schema.AVRO(Example.class))
+                .topic(topic).create();
+```
+
+#### JSON Schema
+The situation with JSON schema is similar to Avro Schema.
+```go
+// Compatible with defining a schema in Java
+exampleSchemaDefCompatible := "{\"type\":\"record\",\"name\":\"Example\",\"namespace\":\"test\"," +
+	"\"fields\":[{\"name\":\"ID\",\"type\":\"int\"},{\"name\":\"Name\",\"type\":[\"null\", \"string\"]}]}"
+
+consumerJSCompatible := NewJSONSchema(exampleSchemaDefCompatible, nil)
+// Not compatible with defining a schema in Java
```

Comment:
Which define code of Java is this incompatible with?

Reply:
Like Avro, is a problem with null fields.